docs: document Tempo instrumentation (#797)
grafana/tempo: docs/tempo/website/operations/monitoring.md

# Monitoring Tempo

Tempo is instrumented to expose metrics, logs and traces.
Additionally, the Tempo repository has a [mixin](https://github.com/grafana/tempo/tree/main/operations/tempo-mixin) that includes a
set of dashboards, rules and alerts.
Together, these can be used to monitor Tempo in production.

## Instrumentation

Tempo is already instrumented with metrics, logs and traces.
These can be collected to observe Tempo.

#### Metrics

Tempo is instrumented with [Prometheus metrics](https://prometheus.io/).
It emits RED metrics for most services and backends.
The [Tempo mixin](#dashboards) provides several dashboards using these metrics.

#### Logs

Tempo emits logs in the `key=value` ([logfmt](https://brandur.org/logfmt)) format.

#### Traces

Tempo uses the [Jaeger Golang SDK](https://github.com/jaegertracing/jaeger-client-go) for tracing instrumentation.

> As of this writing, only the read path of tempo is instrumented for tracing.

Reviewer: pieces of the write path are traced. mainly the ingesters? unsure if this is worth mentioning or if it will just complicate things.

Author: Might be worth mentioning, but how do you find these traces? The query path posts a helpful request summary, afaik the write path doesn't do this.

Author: Hmm, using automatic logging I see some traces related to `tempopb.Pusher/PushBytes` and `CAS` (not sure what this is?)

Author: I've updated the wording a bit.

Reviewer: yeah, i think we're tracing .1% of our write path starting at the ingesters. so i think tracing is poor enough on the write path it's probably not worth mentioning.

`CAS` may be the ring check and set? Maybe ring interactions are traced? I haven't dug into that.

The tracer can be configured [using environment variables](https://github.com/jaegertracing/jaeger-client-go#environment-variables).

> In general, to enable tracing either set `JAEGER_AGENT_HOST` and `JAEGER_AGENT_PORT`, or `JAEGER_ENDPOINT`.

Reviewer 2: ```suggestion
To enable tracing, set one of the following: `JAEGER_AGENT_HOST` and `JAEGER_AGENT_PORT`, or `JAEGER_ENDPOINT`.
```

The Jaeger client uses remote sampling by default, if the management server is not available no traces will be sent.

> To always send traces (no sampling), the following environment variables can be set:

Reviewer 2: ```suggestion
To always send traces (no sampling), set the following environment variables:
```

```
JAEGER_SAMPLER_TYPE=const
JAEGER_SAMPLER_PARAM=1
```

## Dashboards

> The [Tempo mixin](https://github.com/grafana/tempo/tree/main/operations/tempo-mixin) has four Grafana dashboards in the `yamls` folder that can be downloaded and imported into your Grafana UI.

Reviewer 2: ```suggestion
The [Tempo mixin](https://github.com/grafana/tempo/tree/main/operations/tempo-mixin) has four Grafana dashboards in the `yamls` folder that you can download and import into your Grafana UI.
```
